src/PlusDataCollection/WinProbe/vtkPlusWinProbeVideoSource.cxx

```diff
@@ -1198,41 +1198,91 @@ int32_t vtkPlusWinProbeVideoSource::GetSpatialCompoundCount()
 }
 
 //----------------------------------------------------------------------------
-
-void vtkPlusWinProbeVideoSource::SetBRFEnabled(bool value)
+PlusStatus vtkPlusWinProbeVideoSource::SetExtraSourceMode(Mode mode)
 {
-  if(Connected)
+  if (m_Mode == mode)
   {
-    if(m_Mode == Mode::M)
-    {
-      SetMIsEnabled(false);
-    }
-    if(m_Mode == Mode::ARFI)
-    {
-      SetARFIEnabled(false);
-    }
-    if(value)
-    {
-      SetHandleBRFInternally(false);
-      SetBFRFImageCaptureMode(2);
-    }
-    else
-    {
-      SetHandleBRFInternally(true);
-      SetBFRFImageCaptureMode(0);
-    }
+    return PLUS_SUCCESS;
   }
```

Comment:
I don't think with the way `m_Mode` currently is that we can reliably say that we want to return right away. We should continue with the rest of the function until we flesh out `vtkPlusWinProbeVideoSource::setMode`

Reply:
This is just saying if the Mode you are trying to set is already the current mode, then you don't need to do anything.

Reply:
right, but we have competing modes here essentially, because `m_Mode` is Plus-side and for example, can be Mode::B while ::GetMIsEnabled returns true potentially. Until we have a single method which addresses changing all the modes I'm worried about doing this

Reply:
I'm unaware of anything in UltraVisionAPI that actually returns current mode information.

Reply:
ok not quite modes, but I guess we have competing state information. the alternative i would be ok with here is to check both the winprobe and plus state information like
```
if (m_Mode == mode)
{
  if (::GetARFIIsEnabled && mode == mode::ARFI) || (::GetBFRFImageCaptureMode() == 2 && mode == mode::BRF) ....
  {
    return PLUS_SUCCESS
  }
}
```